R/description.R

```diff
@@ -1,38 +1,48 @@
-#' Create a default DESCRIPTION file for a package.
+#' Create a DESCRIPTION file
 #'
 #' @description
-#' If you create a lot of packages, you can override the defaults by setting
-#' option `"usethis.description"` to a named list. Here's an example of code one
-#' could include in `.Rprofile`:
+#' usethis consults the following sources, in this order, to set DESCRIPTION
+#' fields:
+#' * `fields` argument of [create_package()] or [use_description()]
+#' * `getOption("usethis.description")` or `getOption("devtools.desc")`. The
+#' devtools option is consulted only for backwards compatibility and it's
+#' recommended to switch to an option named "usethis.description".
+#' * Defaults built into usethis
+#'
+#' If you create a lot of packages, consider storing personalized defaults as a
+#' named list in an option named `"usethis.description"`. Here's an example of
+#' code to include in `.Rprofile`:
 #'
 #' ```
 #' options(
 #'   usethis.name = "Jane Doe",
 #'   usethis.description = list(
-#'     `Authors@R` = 'person("Jane", "Doe", email = "jane@example.com", role = c#' ("aut", "cre"))',
+#'     `Authors@R` = 'person("Jane", "Doe", email = "jane@example.com", role = c("aut", "cre"))',
 #'     License = "MIT + file LICENSE",
-#'     Version = "0.0.0.9000"
+#'     Language: es
 #'   )
 #' )
 #' ```
 #'
-#' @param fields A named list of fields to add to \file{DESCRIPTION},
-#'   potentially overriding the defaults. If `NULL`, retrieved from
-#'   `getOption("usethis.description")`, and (for backward compatibility) from
-#'   `getOption("devtools.desc")`.
+#' @param fields A named list of fields to add to DESCRIPTION, potentially
+#'   overriding default values. See [use_description()] for how you can set
+#'   personalized defaults using package options
 #' @export
 #' @examples
 #' \dontrun{
 #' use_description()
+#'
+#' use_description(fields = list(Language = "es"))
 #' }
 use_description <- function(fields = NULL) {
   name <- project_name()
   check_package_name(name)
 
-  fields <- fields %||%
-    getOption("usethis.description") %||%
-    getOption("devtools.desc") %||%
-    list()
+  ## the definitive source of user-supplied info: in this call or via options
+  fields <- utils::modifyList(
```

Comment:
It might be worth doing some light type checking here - e.g. that it's a list with names?

Reply:
And should all this logic be moved into `build_description()`? I think that would make the tests simpler since you wouldn't need so many temporary projects

Reply:
I'll try to move things around.

Part of the reason I tested via `use_description()`, though, was to test that directly specified fields override options & defaults, but in a targetted manner.

Reply:
Yeah, I think it would be better to move all that code out so you can test independently of the writing.